lib/container.rb

```diff
@@ -51,12 +52,48 @@ def chmod_r(path, mode)
     run(%Q{lxc exec #{@name} -- chmod -R 0#{mode.to_s(8)} "#{path}"})
   end
 
-  def exec_driver(path, secret, sub, gra)
-    push(path, "/root/bn_driver.rb")
-    command = %Q{lxc exec "#{@name}" -- bash -c \
-                   '(cd && BN_KEY="#{secret}" BN_SUB="#{sub}" BN_GRA="#{gra}" \
-                   ruby -I /tmp/bn/lib bn_driver.rb)'}
-    puts "Run driver: #{command}"
+  def exec_driver(secret, sub_up, gra_up, xtr_up = nil)
+    puts "XX In exec_driver"
+
+    FileUtils.mkdir_p("/tmp/bn")
+    Tempfile.create("driver.", "/tmp/bn") do |drv|
+      puts "XX in block"
```

Comment:
What's with these "in block" comments, and where did block 3 go, and why can't they all go away?